namer/namer.cc

```diff
@@ -516,8 +517,33 @@ class NameInserter {
         return data->intrinsic != nullptr && data->resultType == nullptr;
     }
 
-    bool paramsMatch(core::MutableContext ctx, core::Loc loc, const vector<ast::ParsedArg> &parsedArgs) {
+    bool paramsMatch(core::MutableContext ctx, core::SymbolRef method, const vector<ast::ParsedArg> &parsedArgs) {
+        auto sym = method.data(ctx)->dealias(ctx);
+        if (sym.data(ctx)->arguments().size() != parsedArgs.size()) {
+            return false;
+        }
+        for (int i = 0; i < parsedArgs.size(); i++) {
+            auto &methodArg = parsedArgs[i];
+            auto &symArg = sym.data(ctx)->arguments()[i];
+
+            if (symArg.flags.isKeyword != methodArg.keyword || symArg.flags.isBlock != methodArg.block ||
+                symArg.flags.isRepeated != methodArg.repeated ||
+                (symArg.flags.isKeyword && symArg.name != methodArg.local._name)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    void paramMismatchErrors(core::MutableContext ctx, core::Loc loc, const vector<ast::ParsedArg> &parsedArgs) {
         auto sym = ctx.owner.data(ctx)->dealias(ctx);
+        auto owner = sym.data(ctx)->owner;
+        string methodName = fmt::format("{}{}{}", owner.show(ctx), owner.data(ctx)->isSingletonClass(ctx) ? "." : "#",
```

Comment:
This replicates some of the `show` logic, but the difference between this and `show` is that this uses `shortName`, which means we won't produce errors that specifically reference mangled names. Otherwise, during incremental operation, we'll have situations where the method that we're referencing has already been mangled by a previous pass and therefore will have a name like `f$1`.

Reply:
This has a few problems:
 - it does a costly computation outside of `if (auto e = ctx.state.beginError(...))`, which is an antipattern
 - why is it special compared to other errors? We should either make show consistently do what this does or keep it it uniform with other errors.